cdap-elastic/src/main/java/io/cdap/cdap/metadata/elastic/ElasticsearchMetadataStorage.java

```diff
@@ -1086,9 +1091,35 @@ private QueryBuilder createTermQuery(String term, String textField, SearchReques
   /**
    * Create a query for a single term in a given field.
    *
-   * @return a wildcard query is the term contains * or ?, or a match query otherwise
+   * @return a range query if the term is a number,
+   * a wildcard query if the term is a string and contains * or ?,
+   * or a match query otherwise
    */
-  private QueryBuilder createTermQuery(String field, String term) {
+  private QueryBuilder createTermQuery(QueryTerm queryTerm, String field, String term) {
+    // if attempting to conduct a numeric search
+    if (field.equals(NESTED_NUMERICVALUE_FIELD)) {
+      try {
+        switch (queryTerm.getComparison()) {
+          case GREATER:
+            return QueryBuilders.rangeQuery(field).gt(Long.parseLong(QueryParser.extractTermValue(term)));
+          case GREATER_OR_EQUAL:
+            return QueryBuilders.rangeQuery(field).gte(Long.parseLong(QueryParser.extractTermValue(term)));
+          case LESS:
+            return QueryBuilders.rangeQuery(field).lt(Long.parseLong(QueryParser.extractTermValue(term)));
+          case LESS_OR_EQUAL:
+            return QueryBuilders.rangeQuery(field).lte(Long.parseLong(QueryParser.extractTermValue(term)));
+          case EQUALS:
+            // emulate an equality search
+            return QueryBuilders.rangeQuery(field)
```

Comment:
can't you just use QueryBuilders.termQuery(field, QueryParser.extractTermValue(term)). I believe that represents (numeric) equality if the type of the field is numeric, which it is.

Reply:
Ah, I didn't realize that. This will be changed in the next commit.